python_modules/dagster/dagster/_grpc/proxy_server.py

```diff
@@ -100,6 +101,15 @@ def __init__(
             daemon=True,
         )
 
+        self.__last_heartbeat_time = time.time()
+        self.__server_heartbeat_thread = threading.Thread(
+            target=self._server_heartbeat_thread,
+            args=(server_heartbeat_timeout,),
+            name="grpc-server-heartbeat",
+            daemon=True,
+        )
+        self.__server_heartbeat_thread.start()
```

Comment:
we should still join() this thread on cleanup / shutdown (even though its a daemon thread, still good to do cleanup)